subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_pn.c

```diff
@@ -119,3 +123,33 @@ int fp_storage_pn_save(const char *pn_to_save)
 
 	return 0;
 }
+
+static int fp_storage_pn_reset(void)
+{
+	int err;
+
+	err = settings_delete(SETTINGS_PN_FULL_NAME);
+	if (err) {
+		return err;
+	}
+
+	memset(personalized_name, 0, sizeof(personalized_name));
+
+	settings_set_err = 0;
+	atomic_set(&settings_loaded, true);
+
+	reset_in_progress = false;
```

Comment:
Maybe we could consider separating function to clear data from RAM? That could improve readability.